src/CombFilter/CombFilter.cpp

```diff
@@ -0,0 +1,145 @@
+// CCombIIRFilter definition
+class CCombIIRFilter : public CCombFilterBase
+{
+public:
+    CCombIIRFilter(int delayLength, int iNumChannels, float gain);
+    ~CCombIIRFilter();
+protected:
+//    int m_iNumChannels;
+    Error_t process (float **ppfInputBuffer, float **ppfOutputBuffer, int iNumberOfFrames) override;
+};
+// =====================================================================================================
+// CCombFIRFilter method definition
+CCombIIRFilter::CCombIIRFilter(int delayLength, int iNumChannels, float gainValue): CCombFilterBase(delayLength, iNumChannels, gainValue)
+{
+}
+Error_t CCombIIRFilter::process (float **ppfInputBuffer, float **ppfOutputBuffer, int iNumberOfFrames)
+{
+    // Implement process method for IIR Filter
+    // IIR LOGIC
+    // 1. Read ring buffer at get pointer (and increment)
+    // 2. Process signal => output = input + gain * read value
+    // 3. Put current output into ring buffer and increment write pointer
+    for(int channel=0;channel < m_iNumChannels;channel++)
+    {
+        for(int i=0;i<iNumberOfFrames;i++)
+        {
+            ppfOutputBuffer[channel][i] = ppfInputBuffer[channel][i] + m_fGainValue * m_RingBuffer[channel]->getPostInc();
+            m_RingBuffer[channel]->putPostInc(ppfOutputBuffer[channel][i]);
+        }
+    }
+    return Error_t::kNoError;
+}
+
+CCombIIRFilter::~CCombIIRFilter()
+{
+}
+
+// =====================================================================================================
+// CCombIIRFilter definition
+class CCombFIRFilter : public CCombFilterBase
+{
+    
+public:
+    CCombFIRFilter(int delayLength, int iNumChannels, float gain);
+    ~CCombFIRFilter();
+
+protected:
+//    int m_iNumChannels;
+    Error_t process (float **ppfInputBuffer, float **ppfOutputBuffer, int iNumberOfFrames) override;
+};
+
+// =====================================================================================================
+// CCombIIRFilter method definition
+
+CCombFIRFilter::CCombFIRFilter(int delayLength, int iNumChannels, float gainValue): CCombFilterBase(delayLength, iNumChannels, gainValue)
+{
+
+}
+Error_t CCombFIRFilter::process (float **ppfInputBuffer, float **ppfOutputBuffer, int iNumberOfFrames)
+{
+    // Implement process method for FIR Filter
+    // FIR LOGIC
+    // 1. Read ring buffer at get pointer (and increment)
+    // 2. Process signal => output = input + gain * read value
+    // 3. Put current input into ring buffer and increment write pointer
+    for(int channel=0;channel < m_iNumChannels;channel++)
+    {
+        for(int i=0;i<iNumberOfFrames;i++)
+        {
+            ppfOutputBuffer[channel][i] = ppfInputBuffer[channel][i] + m_fGainValue * m_RingBuffer[channel]->getPostInc();
+            m_RingBuffer[channel]->putPostInc(ppfInputBuffer[channel][i]);
+        }
+    }
+    return Error_t::kNoError;
+}
+
+CCombFIRFilter::~CCombFIRFilter()
+{
+}
```

Comment:
No delete in destructor for heap memory newed in constructor. Will cause memory leak.